src/app/clusters/boolean-state-configuration-server/boolean-state-configuration-server.cpp

```diff
@@ -0,0 +1,391 @@
+#include <lib/core/CHIPError.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/CHIPDeviceConfig.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::BooleanStateConfiguration::Attributes;
+using chip::app::Clusters::BooleanStateConfiguration::Delegate;
+using chip::Protocols::InteractionModel::Status;
+
+static constexpr size_t kBooleanStateConfigurationDelegateTableSize =
+    EMBER_AF_BOOLEAN_STATE_CONFIGURATION_CLUSTER_SERVER_ENDPOINT_COUNT + CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT;
+
+static_assert(kBooleanStateConfigurationDelegateTableSize <= kEmberInvalidEndpointIndex,
+              "BooleanStateConfiguration Delegate table size error");
+
+namespace {
+Delegate * gDelegateTable[kBooleanStateConfigurationDelegateTableSize] = { nullptr };
+
+Delegate * GetDelegate(EndpointId endpoint)
+{
+    uint16_t ep = emberAfGetClusterServerEndpointIndex(endpoint, BooleanStateConfiguration::Id,
+                                                       EMBER_AF_BOOLEAN_STATE_CONFIGURATION_CLUSTER_SERVER_ENDPOINT_COUNT);
+    return (ep >= kBooleanStateConfigurationDelegateTableSize ? nullptr : gDelegateTable[ep]);
+}
+
+bool isDelegateNull(Delegate * delegate, EndpointId endpoint)
+{
+    if (delegate == nullptr)
+    {
+        return true;
+    }
+    return false;
+}
+
+class BooleanStateConfigAttrAccess : public AttributeAccessInterface
+{
+public:
+    BooleanStateConfigAttrAccess() : AttributeAccessInterface(Optional<EndpointId>::Missing(), BooleanStateConfiguration::Id) {}
+
+    CHIP_ERROR Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder) override;
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+private:
+    CHIP_ERROR WriteCurrentSensitivityLevel(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder);
+};
+
+BooleanStateConfigAttrAccess gAttrAccess;
+
+CHIP_ERROR BooleanStateConfigAttrAccess::WriteCurrentSensitivityLevel(const ConcreteDataAttributePath & aPath,
+                                                                      AttributeValueDecoder & aDecoder)
+{
+    uint8_t curSenLevel, supportedSensLevel;
+    ReturnErrorOnFailure(aDecoder.Decode(curSenLevel));
+    VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == SupportedSensitivityLevels::Get(aPath.mEndpointId, &supportedSensLevel),
+                        CHIP_IM_GLOBAL_STATUS(Failure));
+
+    VerifyOrReturnError(curSenLevel < supportedSensLevel, CHIP_IM_GLOBAL_STATUS(ConstraintError));
+    VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == CurrentSensitivityLevel::Set(aPath.mEndpointId, curSenLevel),
+                        CHIP_IM_GLOBAL_STATUS(Failure));
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR BooleanStateConfigAttrAccess::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    return CHIP_NO_ERROR;
```

Comment:
If we do keep the AttributeAccessInterface, this should document why fallthrough to the underlying attribute store is fine: we are using that in practice to store the attribute value.

Reply:
I've stopped using attribute store now.